src/hooks/useFileUpload.tsx

```diff
@@ -300,22 +317,15 @@ export default function useFileUpload(
         return;
       }
     }
-    for (const [index, file] of files.entries()) {
-      const filename =
-        allowNameFallback && uploadFileNames[index] === "" && file
-          ? file.name
-          : uploadFileNames[index];
-      if (!filename) {
-        setError(t("file_error__single_file_name"));
-        return;
-      }
-      setUploading(true);
 
+    setUploading(true);
+
+    for (const [index, file] of files.entries()) {
       try {
         const data = await createUpload({
           original_name: file.name ?? "",
           file_type: fileType,
-          name: filename,
+          name: uploadFileNames[index],
```

Comment:
```suggestion
          name:  allowNameFallback && uploadFileNames[index] === "" && file
            ? file.name
            : uploadFileNames[index];,
```

Reply:
Also have useFileUpload return setError as well. That way, you can reset the error after user enters the filename.